src/d3/hapticplot.d3.spec.ts

```diff
@@ -1,55 +1,101 @@
+    expect(result).toEqual(expectedColorArray);
+  });
+
+  it('places points for each datum and sets the correct size property on the resulting shape entities', () => {
+    hapticplot.init(scene, [10, 20, 30]);
+    const expectedSizeArray = ['0.05', '0.05', '0.05'];
+    const result = getRadius(scene, shape);
+    expect(result).toEqual(expectedSizeArray);
+  });
+
+  it('places points for each datum and sets the correct hoverable property on the resulting shape entities', () => {
+    hapticplot.init(scene, [10, 20, 30]);
+    const expectedSizeArray = ['hoverable', 'hoverable', 'hoverable'];
+    const result = getHoverable(scene, shape);
+    expect(result).toEqual(expectedSizeArray);
+  });
+
+  it('places points for each datum and sets the correct color property on the resulting shape entities after a hover event', () => {
+    hapticplot.init(scene, [10, 20, 30]);
+    const expectedSizeArray = ['red', 'red', 'red'];
+    const result = getHoveredColor(scene, shape);
+    expect(result).toEqual(expectedSizeArray);
   });
 });
 
+// Helper Functions
+
 // returns array of actual position vectors
-function getPosition(element: HTMLElement, shape: string): Array<{x: number, y: number, z: number}>{
-  const childrenArray = element.querySelectorAll(shape);
-  const positionArray: Array<{x: number, y: number, z: number}> = [];
-  for (const child of (childrenArray as any)){
-    positionArray.push((child as any).components.position.attrValue);
-  }
-  return positionArray;
+function getPosition(scene: HTMLElement, shape: string): Vector3[]{
+  const attrArray: Vector3[] = [];
+  Array.from(scene.querySelectorAll(shape)).forEach((child) => {
+    attrArray.push(new Vector3(
+      (child as Entity).object3D.position.x,
+      (child as Entity).object3D.position.y,
+      (child as Entity).object3D.position.z));
+  });
+  return attrArray;
+}
+
+// returns array of each generated objects color
+function getColor(scene: HTMLElement, shape: string): (string | null)[]{
+  return Array.from(scene.querySelectorAll(shape)).map((point: Element) => point.getAttribute('color'));
+}
+
+// returns array of each generated objects radius
+function getRadius(scene: HTMLElement, shape: string): (string | null)[]{
+  return Array.from(scene.querySelectorAll(shape)).map((point: Element) => point.getAttribute('radius'));
+}
+
+// returns array of each generated objects hover property
+function getHoverable(scene: HTMLElement, shape: string): (string | undefined)[]{
+  const attrArray: (string | undefined)[] = [];
+  Array.from(scene.querySelectorAll(shape)).forEach((child) => { attrArray.push((child as Entity).components.hoverable.attrName); });
+  return attrArray;
+}
+
+// returns array of each generated objects color, after a hover event has occured
+function getHoveredColor(scene: HTMLElement, shape: string): (string | null)[]{
+  Array.from(scene.querySelectorAll(shape)).forEach((child) => { child.dispatchEvent(new Event('hover-start')); });
```

Comment:
This line would probably be easier to read as a for..of loop.  Google's styleguide actually forbids `forEach` entirely, but in this case in particular it would dodge the `Array.from`

Reply:
I changed it to a for...of loop, but I'm still using Array.from since the NodeListOf<Element> returned by the querySelector has no iterator. Is there a better way to do this?